packages/toc/src/treeview.tsx

```diff
@@ -26,8 +28,18 @@ export class TableOfContentsWidget extends VDomRenderer<TableOfContents.IModel<T
    * includes layout triggered rendering.
    */
   render(): JSX.Element | null {
-    if (!this.model) {
-      return null;
+    if (!this.model || this.model.headings.length === 0) {
+      this.parent?.addClass('jp-TableOfContents-empty');
+      return (
+        <div className="jp-TableOfContents-placeholder">
+          <div className="jp-TableOfContents-placeholderContent">
+            <h1>{this._placeholderHeadline}</h1>
```

Comment:
```suggestion
            <h3>{this._placeholderHeadline}</h3>
```

`<h1>` is out of place here (the hierarchy should be maintained) - it cannot be after `<h2>` in hierarchy (and sidebar title is `<h2>`). It also looks too big and IMO should be in grey and use standard themed sizes and colour (as in the Brian's point about using lab theme variables).